File: HelperJobby/DTOs/UserJobInteractions/JobApplyDTO.cs
using ApplicationDomain.Enums;
using HelperJobby.DTOs.Account;
using HelperJobby.DTOs.Job;

namespace HelperJobby.DTOs.UserJobInteractions;

public class JobApplyDTO
{
    public int JobId { get; set; }

    public int JobSeekerId { get; set; }
    public JobApplyStatuses JobApplyStatus { get; set; }
    public JobDTO Job { get; set; }

    public JobSeekerDTO JobSeeker { get; set; }

    public DateOnly DateApplied { get; set; }
}

File: HelperJobby/client-app/src/DTOs/userJobInteractionsDTOs/JobApplyDTO.ts
import {JobSeekerDTO} from "../accountDTOs/JobSeekerDTO";
import {JobDTO} from "../jobRelatetedDTOs/JobDTO";
import {JobApplyStatuses} from "../../enums/modelDataEnums/JobApplyStatuses";
import {ResumeDTO} from "../resumeRelatedDTOs/ResumeDTO";

export interface JobApplyDTO {
    jobId: number;
    jobSeekerId: number;
    job: JobDTO;
    jobSeeker: JobSeekerDTO;
    resume : ResumeDTO;
    dateApplied: string;
    jobApplyStatus : JobApplyStatuses;
    isReviewed : boolean;
}